optimade/models/utils.py

```diff
@@ -182,6 +183,22 @@ def base_version(self) -> str:
         return f"{self.major}.{self.minor}.{self.patch}"
 
 
+def anonymous_element_generator():
+    """ Generator that yields the next symbol in the A, B, Aa, ... Az naming scheme. """
+    from string import ascii_lowercase
+
+    for size in itertools.count(1):
+        for s in itertools.product(ascii_lowercase, repeat=size):
+            s = list(s)
+            s[0] = s[0].upper()
+            yield "".join(s)
+
+
+ANONYMOUS_ELEMENTS = tuple(itertools.islice(anonymous_element_generator(), 150))
+""" Returns the first 150 values of the anonymous element generator. """
```

Comment:
I think I decided that recreating this for every structure could get quite expensive, and doing the full infinite generator could bring down a server and has no sensible use-case

Reply:
The use-case this would guard against would be a structure containing 702 different elements (150 already seemed like overkill to me) so I think I'll leave this as it is...